src/classmentors/services.js

```diff
@@ -1349,11 +1350,11 @@ export function clmDataStoreFactory(
       },
 
       _getRanking: function(data) {
-        var ranking = {
-          singPath: clmDataStore.events._solvedProblems(data.singPath),
-          codeCombat: data.badges.codeCombat.length,
-          codeSchool: data.badges.codeSchool.length
-        };
+        var ranking = {singPath: clmDataStore.events._solvedProblems(data.singPath)};
+
+        Object.keys(data.userAchievements || {}).forEach(serviceId => {
+          ranking[serviceId] = data.userAchievements[serviceId].totalAchievements || 0;
+        });
```

Comment:
Just realised ranking is not pre-calculated anymore. I will just remove that part.